segment.go

```diff
@@ -56,6 +56,8 @@ const (
 	ShellInfo SegmentType = "shell"
 	//Node writes which node version is currently active
 	Node SegmentType = "node"
+	//OS write os specific icon
+	OsIcon SegmentType = "osicon"
```

Comment:
Can we rename this to Os? As people will be able to set any string (which could also be plain `"windows"` for example), it feels more natural that way. I would leave `WindowsIcon` etc as is as the other segments use the same context and most likely people will be adding an icon.

Reply:
I originally had it as "os", but had a naming collision with the os pkg.  I'll try changing the segment name to os and call the internal struct something else.

Reply:
That's the best solution indeed 👍🏻